src/Analyzers/Core/Analyzers/IDEDiagnosticIdToOptionMappingHelper.cs

```diff
@@ -21,12 +21,16 @@ internal static class IDEDiagnosticIdToOptionMappingHelper
     {
         private static readonly ConcurrentDictionary<string, ImmutableHashSet<IOption2>> s_diagnosticIdToOptionMap = new();
         private static readonly ConcurrentDictionary<string, ConcurrentDictionary<string, ImmutableHashSet<IOption2>>> s_diagnosticIdToLanguageSpecificOptionsMap = new();
+        private static readonly ConcurrentDictionary<string, PerLanguageOption2<bool>> s_diagnosticIdToFadingOptionMap = new();
```

Comment:
this all being static data isn't great.  but i can see the precedence, and it would be a lot to change this (esp. at the analyzers layer) so ok to keep for now.